wodles/aws/buckets_s3/waf.py

```diff
@@ -56,3 +83,30 @@ def json_event_generator(data):
                         sys.exit(9)
 
         return json.loads(json.dumps(content))
+
+    def get_service_prefix(self, account_id):
+        return AWSLogsBucket.get_service_prefix(self, account_id)
+        
+    def get_full_prefix(self, account_id, account_region, acl_name=None):
+        if self.type == WAF_NATIVE:
+            path = self.client.list_objects_v2(Bucket=self.bucket)
+            if 'Contents' in path:
+                for obj in path['Contents']:
+                    log_key = obj['Key']
+                parts = log_key.split("/")
```

Comment:
Just to confirm: is this right? This is looping all the items inside `path['Contents']`, but in each iteration the value of `log_key` is overwritten, so `parts = log_key.split("/")` is only done on the last item of the list.